packages/data-table-server/src/dataTable/AbstractDataTable.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { TupaiaApiClient } from '@tupaia/api-client';
+import { yup } from '@tupaia/utils';
+
+export abstract class AbstractDataTable<
```

Comment:
I know we don't usually do the whole `Abstract...` naming convention. Originally this was just called `DataTable`, but the name clashed with the model classes, which there are already a couple of, so I felt this made it a bit clearer.